// field-descriptions/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FD = exports.GenMD = void 0;
const lodash_1 = require("lodash");
const underline = "__";
const strikethrough = "~~";
const italic = "*";
const bold = "**";
const GenMDOptionsDefault = {
    underline: false,
    strikethrough: false,
    italic: false,
    bold: false,
};
const edits = {
    underline,
    strikethrough,
    italic,
    bold,
};
function reverseString(str) {
    return str.split("").reverse().join("");
}
function GenMD(text, options = GenMDOptionsDefault) {
    const sep = lodash_1.keys(lodash_1.pickBy(options))
        .map((value) => {
        return edits[value];
    })
        .join("");
    return sep + text + reverseString(sep);
}
exports.GenMD = GenMD;
function FD(values, options = {
    nameEdits: GenMDOptionsDefault,
    valueEdits: GenMDOptionsDefault,
}) {
    const { separator, nameEdits, valueEdits } = options;
    return values
        .map(({ name, value }) => {
        return (GenMD(name, nameEdits) +
            (separator ? separator : ":") +
            " " +
            GenMD(value, valueEdits));
    })
        .join("\n");
}
exports.FD = FD;
